src/components/actions/filters/ItemFilterReducers.ts

```diff
@@ -0,0 +1,57 @@
+import { IReduxAction } from '../../../utils/ReduxUtils';
+import { IReduxActionsPayload } from '../../../ReactVapor';
+import { ItemFilterActions } from './ItemFilterActions';
+import * as _ from 'underscore';
+
+export interface IItemFilterState {
+  id: string;
+  item: string;
+}
+
+export const itemFilterOriginalState: IItemFilterState = {
+  id: undefined,
+  item: ''
+};
+
+export const itemFiltersOriginalState: IItemFilterState[] = [];
+
+export const itemFilterReducer = (state: IItemFilterState = itemFilterOriginalState,
+  action: IReduxAction<IReduxActionsPayload>): IItemFilterState => {
+  switch (action.type) {
+    case ItemFilterActions.add:
+      return {
+        id: action.payload.id,
+        item: state.item
+      };
+    case ItemFilterActions.filter:
+      if (state.id !== action.payload.id) {
```

Comment:
Like you said, I'm not sure it is good practice either. It is much clearer this way.